lib/QueryRunner.js

```diff
@@ -45,6 +47,12 @@ export default class QueryRunner {
       })
     }
 
+    const cachedPort = this.portCache[`${userOptions.address}:${userOptions.port}`]
+
+    if (cachedPort && optionsCollection.portCache) {
+      addAttemptWithPort(cachedPort)
```

Comment:
If the cached port fails this will result in 2 attempts to the same port.

Reply:
In the case whereas the server query port changes during runtime, right?

Reply:
It depends on the retry options, but would only occur if there was a timeout or other failure on the cached port yes.